test/decl/func/default-values-swift4.swift

```diff
@@ -111,3 +111,29 @@ public func evilCode(
     }
     return 0
   }()) {}
+
+public struct HasSubscript {
+  public subscript(x: Int = {
+    struct Nested {}
+    // expected-error@-1 {{type 'Nested' cannot be nested inside a default argument value}}
+
+    publicFunction()
+
+    versionedFunction()
+    // expected-error@-1 2{{global function 'versionedFunction()' is internal and cannot be referenced from a default argument value}}
+
+    internalFunction()
+    // expected-error@-1 2{{global function 'internalFunction()' is internal and cannot be referenced from a default argument value}}
+
+    fileprivateFunction()
+    // expected-error@-1 2{{global function 'fileprivateFunction()' is fileprivate and cannot be referenced from a default argument value}}
+
+    privateFunction()
+    // expected-error@-1 2{{global function 'privateFunction()' is private and cannot be referenced from a default argument value}}
+
+    return 0
+  }()) -> Int {
```

Comment:
Am I paranoid for wondering if there might be a difference when not using a closure?

Reply:
Done

Reply:
! That was a big improvement, but what I actually meant was "is it possible usable-from-inline checking works in the closure case but not in the `arg: Int = privateFunction()` case".

Reply:
Look at the last test case in this file :)

Reply:
Whoops, paid too much attention to the rest of it!